build/Shared/HashCodeCombiner.cs

```diff
@@ -83,10 +76,11 @@ internal void AddSequence<T>(IEnumerable<T> sequence)
         {
             if (dictionary != null)
             {
+                CheckInitialized();
                 foreach (var pair in dictionary.OrderBy(x => x.Key))
                 {
-                    AddObject(pair.Key);
-                    AddObject(pair.Value);
+                    AddInt32(pair.Key.GetHashCode());
```

Comment:
Use `CheckInitialized` once at start; then call `AddInt32` which doesn't have the check